modules/internal/DefaultRectangular.chpl

```diff
@@ -2264,10 +2264,17 @@ module DefaultRectangular {
     return res;
   }
 
+  // <=> implementation uses Reflection to check whether this is implemented, so
+  // it is OK for this to not resolve without debugOptimizedSwap.
+  proc DefaultRectangularArr.doiOptimizedSwap(other) where debugOptimizedSwap {
+    writeln("DefaultRectangularArr doing unoptimized swap. Type mismatch");
+    return false;
+  }
```

Comment:
Similar comments here to the BlockArr overloads, obviously...  Whatever we do, we should do for both similarly.